src/vsr/replica.zig

```diff
@@ -2700,20 +2696,17 @@ pub fn ReplicaType(
         }
 
         /// Choose a different replica each time if possible (excluding ourself).
+        ///
+        /// Currently this picks the target replica at random instead of doing something like
+        /// round-robin in order to avoid a resonance.
         fn choose_any_other_replica(self: *Self) u8 {
             assert(!self.solo());
+            comptime assert(constants.replicas_max * 2 < std.math.maxInt(u8));
 
-            var count: usize = 0;
-            while (count < self.replica_count) : (count += 1) {
-                self.choose_any_other_replica_ticks += 1;
-                const replica = @mod(
-                    self.replica + self.choose_any_other_replica_ticks,
-                    self.replica_count,
-                );
-                if (replica == self.replica) continue;
-                return @intCast(u8, replica);
-            }
-            unreachable;
+            const shift = self.prng.random().intRangeAtMost(u8, 1, self.replica_count - 1);
+            const other_replica = @mod(self.replica + shift, self.replica_count);
```

Comment:
:thinking: If we are a standby, this might not send a request to the (active) replica at `self.replica % self.replica_count`.

Reply:
Agonized between writing a retry loop, which naturally handles both casese but looks non-minimal, writing an explicit `if (self.standby())` which looks ugly, and writing cryptic modulo arithmetic, which happens to work for standby case, picked the last solution in the end. 

To bad it's hard to assert that we can pick _any_ replica here.